### PR TITLE
Changes to packaging docs

### LINKED ISSUE #261: Packaging page needs clarification
* Demote cibuildwheel to amber and separate into a "Building" table with build.
* Add mentions of twine, bump2version + forks (red... because it's superseded by setuptools_scm).

Also add a section called "Package configuration file"
* pyproject.toml (green)
* setup.py
* setup.cfg

### PR DESCRIPTION
**Pull request by ruaridhg**
Closes #261

### CODE REVIEW

File: docs/pages/packaging.md [hunk 1]
```diff
@@ -5,12 +5,29 @@ layout: default
...
+| [bump-my-version](https://github.com/callowayproject/bump-my-version) | Tool for version-bumping your software. Superseded by setuptools_scm                                                        | 🔴  |                                                 |
+
+# Building
+
+| Name                                                  | Short description                                                                                         | 🚦  |                         Used by                          |
+| ----------------------------------------------------- | --------------------------------------------------------------------------------------------------------- | --- | :------------------------------------------------------: |
+| [build](https://pypa-build.readthedocs.io/en/stable/) | Straightforward tool to build a Python package.                                                           | 🟢  |                                                          |
+| [cibuildwheel](https://cibuildwheel.readthedocs.io)   | Builds python wheels for the main operating systems on continuous integration runs (e.g. GitHub actions). | 🟠  | [streamtracer](https://github.com/dstansby/streamtracer) |
```
Comment (dstansby) [inline docs/pages/packaging.md]: What's the reason for cibuildwheen going from green to orange?

File: docs/pages/packaging.md [hunk 2]
```diff
@@ -5,12 +5,29 @@ layout: default
...
+
+## Package configuration file
```
Comment (dstansby) [inline docs/pages/packaging.md:24]: 👏 this is an excellent section

File: docs/pages/packaging.md [hunk 3]
```diff
@@ -5,12 +5,29 @@ layout: default
...
+
+| Name                                                                                               | Short description                                                                               | 🚦  | Used by |
+| -------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------- | --- | :-----: |
+| [pyproject.toml](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/)             | Contains build system requirements and information, which are used by pip to build the package. | 🟢  |         |
+| [setup.py](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/)  | Strongly coupled with setuptools and is not recommended                                         | 🟠  |         |
```
Comment (dstansby) [inline docs/pages/packaging.md]:
```suggestion
| [setup.py](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/)  | Strongly coupled with setuptools and therefore not recommended.                                         | 🟠  |         |
```

File: docs/pages/packaging.md [hunk 4]
```diff
@@ -5,12 +5,29 @@ layout: default
...
+| [setup.cfg](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/) | An ini file that contains option defaults for setup.py commands                                 | 🟠  |         |
```
Comment (dstansby) [inline docs/pages/packaging.md]:
```suggestion
| [setup.cfg](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/) | An ini file that contains defaults for setup.py commands.                                 | 🟠  |         |
```

File: docs/pages/packaging.md (hunk 1 above)
Comment (ruaridhg) [reply docs/pages/packaging.md]: Difficult to use according to @paddyroddy

File: docs/pages/packaging.md [hunk 5]
```diff
@@ -5,12 +5,29 @@ layout: default
...
+| [build](https://pypa-build.readthedocs.io/en/stable/) | Straightforward tool to build a Python package.                                                           | 🟢  |                                                          |
+| [cibuildwheel](https://cibuildwheel.readthedocs.io)   | Builds python wheels for the main operating systems on continuous integration runs (e.g. GitHub actions). | 🟠  | [streamtracer](https://github.com/dstansby/streamtracer) |
+
+## Package configuration file
+
+| Name                                                                                               | Short description                                                                               | 🚦  | Used by |
+| -------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------- | --- | :-----: |
+| [pyproject.toml](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/)             | Contains build system requirements and information, which are used by pip to build the package. | 🟢  |         |
```
Comment (samcunliffe) [inline docs/pages/packaging.md]:
```suggestion
| [pyproject.toml](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/)             | Contains build system requirements and information, which are used by pip to build the package. It is becoming the accepted standard and we strongly recommend it. | 🟢  |         |
```

Comment (samcunliffe) [reply docs/pages/packaging.md]: Just to make it clear that "it's not strongly coupled to ..." is not the main reason to use pyproject.toml over the other guys.

File: docs/pages/packaging.md [hunk 6]
```diff
@@ -5,12 +5,29 @@ layout: default
...
+
+## Package configuration file
+
+| Name                                                                                               | Short description                                                                               | 🚦  | Used by |
+| -------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------- | --- | :-----: |
+| [pyproject.toml](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/)             | Contains build system requirements and information, which are used by pip to build the package. It is becoming the accepted standard and we strongly recommend it. | 🟢  |         |
+| [setup.py](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/)  | Strongly coupled with setuptools and therefore not recommended.                                         | 🟠  |         |
```
Comment (t-young31) [inline docs/pages/packaging.md]: As far as I'm aware sometimes (e.g. with `ext_modules`) a setup.py is unavoidable

Comment (paddyroddy) [reply docs/pages/packaging.md]: Yeah, I think it depends on what kind of bindings you have. Pretty sure with `rust` you can avoid it. Also depends on build system of course.

Comment (dstansby) [reply docs/pages/packaging.md]: So is the suggestion here that `setup.py` should go green? Or that there should be an update to the text and it should stay orange?

Comment (t-young31) [reply docs/pages/packaging.md]: 🟠 / 🟢  depending on whether you need it or not 😅

Comment (samcunliffe) [reply docs/pages/packaging.md]: To stop this from going even more stale (it's already months old), I've merged. Please feel encouraged to reopen if there is a wording suggestion here.

**Review by samcunliffe — APPROVED** [review]

**Pull request merged**